javalin/src/main/java/io/javalin/plugin/rendering/vue/JavalinVue.kt

```diff
@@ -102,6 +108,28 @@ object PathMaster {
     }
 }
 
+object FileInliner {
+    private val newlineRegex = Regex("\\r?\\n")
+    private val unconditionalRegex = Regex("""@inlineFile\(".*"\)""")
+    private val devRegex = Regex("""@inlineFileDev\(".*"\)""")
+    private val notDevRegex = Regex("""@inlineFileNotDev\(".*"\)""")
+
+    fun String.inlineFiles(paths: Set<Path>): String {
+        val pathMap = paths.filterNot { it.toString().endsWith(".vue") } // vue files are inlined in @componentRegistration later
+                .associateBy { """"/vue${it.toString().replace("\\", "/").split("/vue")[1]}"""" } // normalize keys
+        return this.split(newlineRegex).joinToString("\n") { line ->
+            if (!line.contains("@inlineFile")) return@joinToString line // nothing to inline
+            val matchingKey = pathMap.keys.find { line.contains(it) } ?: throw IllegalStateException("Invalid path found: $line")
+            val matchingFileContent by lazy { pathMap[matchingKey]!!.readText() }
+            when {
+                devRegex.containsMatchIn(line) -> if (JavalinVue.isDev) line.replace(devRegex, matchingFileContent) else ""
+                notDevRegex.containsMatchIn(line) -> if (!JavalinVue.isDev) line.replace(notDevRegex, matchingFileContent) else ""
+                else -> line.replace(unconditionalRegex, matchingFileContent)
+            }
```

Comment:
Do you need this `when`? Is there ever a case where more than 1 matches?

```suggestion
            if (JavalinVue.isDev) line.replace(devRegex, matchingFileContent) else ""
            if (!JavalinVue.isDev) line.replace(notDevRegex, matchingFileContent) else ""
            line.replace(unconditionalRegex, matchingFileContent)
```

Reply:
I guess maybe for returning `""` in the case where `containsMatchIn(line)` is true, but where you want to return `""`?

Reply:
I guess not, but it feels a bit weird to run one pointless replace each time 🤔

Reply:
It might not look it, but `this.split(newlineRegex).joinToString("\n") { ... }` is a mapping function, and the `when` yields the return value.